src/workerd/api/node/zlib-util.h

```diff
@@ -210,24 +219,44 @@ class ZlibUtil final: public jsg::Object {
         int level,
         int memLevel,
         int strategy,
-        kj::Array<kj::byte> writeState,
+        jsg::BufferSource writeState,
         jsg::Function<void()> writeCallback,
         jsg::Optional<kj::Array<kj::byte>> dictionary);
+    template <bool async>
+    void write_(jsg::Lock& js,
+        int flush,
+        jsg::Optional<kj::Array<kj::byte>> input,
+        int inputOffset,
+        int inputLength,
+        kj::ArrayPtr<kj::byte> output,
+        int outputOffset,
+        int outputLength);
+
+    // TODO(soon): Find a way to expose functions with templates using JSG_METHOD.
```

Comment:
Give this a try:

JSG_METHOD_NAMED(write, template write_<true>)
JSG_METHOD_NAMED(writeSync, write_<false>)

Reply:
unfortunately, it also doesn't work

Reply:
Ah you also need an explicit instantiation of the template, to tell C++ to actually generate the code. Looks kind of clunky but it would be something like:

```c++
template void ZlibUtil::ZlibStream::write_<true>(all your params here);
template void ZlibUtil::ZlibStream::write_<false>(all your params here);
```

Reply:
yeah, i'll leave it to a different pull-request for now.